### PR TITLE
Python3 support

### PR DESCRIPTION
**Pull request by mslehto**
By decoding mail headers and UID counter, I got this plugin to work nice with Python 3.5.1, I try to test with 2.7 soon as well.

Without UID decode IMAP SEARCH command looked something like this:
DFOI3 SEARCH (UID b'2':*)

### CODE REVIEW

File: mailwatch.py [hunk 1]
```diff
@@ -83,15 +83,23 @@ def poll(self, host, mailbox, user, passwd, room, ssl=True):
...
-				message += "\n\tCc: {0}".format(email.header.decode_header(mail.get('cc'))[0][0])
-				message += "\n\tSubject: {0}".format(email.header.decode_header(mail.get('subject'))[0][0])
+
+				message = ''
+				for hdrname in ['From','To','Cc','Subject']:
+					value = mail.get(hdrname) or None
+					if value:
+						(hdrvalue, charset) = email.header.decode_header(value)[0]
+						if isinstance(hdrvalue, str):
+							message += "{}: {}\n".format(hdrname, hdrvalue)
```
Comment (zoni) [inline mailwatch.py]: The original message put tabs (`\t`) in front of these lines to make it a little easier to visually distinguish multiple messages when there were multiple new messages. Would you mind adding the tabs back in?

File: mailwatch.py [hunk 2]
```diff
@@ -83,15 +83,23 @@ def poll(self, host, mailbox, user, passwd, room, ssl=True):
...
+						elif charset is not None:
+							message += "{}: {}\n".format(hdrname, hdrvalue.decode(charset,'ignore'))
```
Comment (zoni) [inline mailwatch.py]: The original message put tabs (`\t`) in front of these lines to make it a little easier to visually distinguish multiple messages when there were multiple new messages. Would you mind adding the tabs back in?

File: mailwatch.py [hunk 3]
```diff
@@ -83,15 +83,23 @@ def poll(self, host, mailbox, user, passwd, room, ssl=True):
...
+						else:
+							message += "{}: {}\n".format(hdrname, hdrvalue)
```
Comment (zoni) [inline mailwatch.py]: The original message put tabs (`\t`) in front of these lines to make it a little easier to visually distinguish multiple messages when there were multiple new messages. Would you mind adding the tabs back in?

**Comment by zoni** [comment]
Heh, I remember hacking this plugin together one fateful afternoon :smile: 

Thanks for going through the effort of trying to make this Python 3 compatible! If you could verify it still runs on Python 2.7 with these changes, that'd be great. I'd be happy to merge it.

If it doesn't run on Python 2 with these changes and you can't make it work with both, I'd still be happy to merge it as a py3-only plugin. I'd rather support py3 than py2.

**Comment by mslehto** [comment]
Sure, formatting is restored.
Header decoding business did not initially work with Python 2, but now does (2.7.10).
Current version also makes lot more sense with Python 3.

**Comment by zoni** [comment]
Thanks!

**Pull request merged**